src/tsx/GamePanel.tsx

```diff
@@ -82,55 +99,55 @@ export function GamePanel () {
     }
   }
 
-  const materialDifference = (fen: string) => {
-    let board = fen.split(' ')[0]
-    if (displayIndex !== 0) {
-      board = displayGame.moves[displayIndex].fen.split(' ')[0]
+  const materialDifference = (fen: string): { white: JSX.Element[], black: JSX.Element[] } => {
+    const board = fen.split(' ')[0]
+    const whitePieces: PieceCount = { q: 0, r: 0, b: 0, n: 0, p: 0 }
+    const blackPieces: PieceCount = { q: 0, r: 0, b: 0, n: 0, p: 0 }
+
+    // Count pieces on the current board
+    for (const char of board) {
+      if (char === '/') continue
+      const num = parseInt(char, 10)
+      if (!isNaN(num)) continue
+
+      const piece = char.toLowerCase()
+      const isWhite = char === char.toUpperCase()
+      const count = isWhite ? whitePieces : blackPieces
+      count[piece]++
```

Comment:
This is quite elegant but looks bananas, better naming and some comments could help.